src/docs/tiered-storage.dox

```diff
@@ -0,0 +1,81 @@
+/*! @page tiered_storage Tiered Storage
+
+\warning The WiredTiger Tiered Storage support has not been developed, tested or tuned in
+production scenarios, and should be viewed as experimental code.
+
+@section tiered_background Background
+
+A tiered table is a single btree (possibly with multiple checkpoints) made
+up of blocks that reside in one or more files on the local file system and
+zero or more cloud objects (e.g., stored in S3). The btree spans this set of
+files and objects by encoding a block's location in the pointer that links
+blocks in the btree. A tiered btree is contrasted with a regular (non-tiered)
+btree that is contained in a single file. It is also different from an LSM
```

Comment:
Is all lower-case `btree` typically used in non-arch docs? Whatever is used this should be consistent.

Reply:
We're going with `btree`, that's been the dominant form.